src/socket-proxy/socket-proxyd.c

```diff
@@ -75,6 +77,26 @@ static void connection_free(Connection *c) {
         free(c);
 }
 
+static int connection_release(Connection *c) {
+        int r;
+        Context *context = c->context;
+
+        connection_free(c);
+
+        if (set_isempty(context->connections)) {
```

Comment:
I'd expect for this to be conditionalized on `arg_exit_idle_time < USEC_INFINITY`. No need to do the operation otherwise.